### DISCUSSION TITLE
Conditional import statement

### DISCUSSION THREAD

**Discussion by janorn** [opentofu/opentofu#2286] [Q&A] [answered]
Hi,

We are trying to use the import function in tofu 1.7.6. However we run into an issue where we want to migrate a local-exec into proper provider while at the same time new servers could be built using same code.

This code will work on an existing server:
```
import {
  to = resource.network
  id = var.net_id
}
```

But this will fail on a new host which doesn't have a pre existing net_id. Is there anyway to make this import conditional if say net_id is undefined?

**Comment by ghost** [comment]
Hi @janorn the `import` block supports `for_each`, you may be able to use it to create a conditional import.

**Reply by janorn** [reply]
I did try with this statement:
```
import {
  for_each = var.import_resource
  to       = each.key
  id       = each.value
}
```
With this input variable:
```
inputs = {
  import_resource = {
    "resource.network" = var.net_id
  }
}
```
I didn't work very well. Don't seem to be possible to use a variable in the to parameter.

**Reply by ghost** [reply]
Can you raise an issue about this use case so we can look into it more thoroughly?

**Reply by janorn** [reply, marked as answer]
Sure but this workaround worked:
```
import {
  for_each = var.import_resource  # Ignore key
  to       = resource.network
  id       = each.value
}
```
With this as input:
```
inputs = {
  import_resource = {
    key = var.net_id
  }
}
```

**Comment by janorn** [comment]
Github [issue](https://github.com/opentofu/opentofu/issues/2290)
